### ISSUE TITLE
Upgrade to tokio 0.3

### ISSUE THREAD

**Issue by CalliEve** [slowtec/tokio-modbus#71]
Hi!

Currently tokio is on version 0.3.6, while this crate only supports the 0.2 versions of tokio.

Using the 0.3 versions of tokio leads to a `thread 'tokio-runtime-worker' panicked at 'there is no reactor running, must be called from the context of Tokio runtime'` runtime error, when using the tcp client to connect to something.

**Comment by flosse** [comment]
@Baev1 Thanks for reporting this. Unfortunately we currently have no capacity to update this crate.
So feel free to open a PR ;-)

**Comment by Baev1** [comment]
@flosse thank you for your reply! 

I will look into making a PR to help with this, but after looking into things a bit more, I will wait for the soon upcoming 1.0 release of tokio (said to come at the end of this month) as announced back in October in their [blog](https://tokio.rs/blog/2020-10-tokio-0-3), as to prevent having to make another PR soon afterwards again.

**Comment by flosse** [comment]
> I will look into making a PR to help with this,

great!
And don't hesitate to ask for help :)

**Comment by uklotzde** [comment]
Afterwards we need to bump the version number to v0.5. Many crates still require tokio 0.2.x which might cause dependency conflicts in a project? Not sure about this.

But because of these potential conflicts we should prepare ourselves for the upgrade. You could start with tokio 0.3.x to determine the current gap. 1.0 was announced to released before the end of this year.

**Comment by Baev1** [comment]
I skimmed through the dependencies and it seems to me `tokio-serial` will be a blocker for this, as it also uses tokio 0.2. In addition tokio-serial uses mio-serial, which depends on mio 0.6, while afaik tokio 0.3 uses mio 0.7 and I am not sure if that would then in turn block tokio-serial from using tokio 0.3 or not. 

I can't with confidence say if this all means we can or can't upgrade tokio here, but yeah it is be likely that I first have to take a look at making a PR for tokio-serial to bump their tokio version.

Also, unrelated, but I noticed that `net2` is deprecated in favour of `socket2`.
